### PR TITLE
Addresed comments from Andrey

### PR DESCRIPTION
**Pull request by orrrrtem**

### CODE REVIEW

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.hpp [hunk 1]
```diff
@@ -22,76 +22,86 @@
...
+
+    // vertex weight types
+    using vertex_user_value_type       = VertexValue;
+    using const_vertex_user_value_type = const vertex_user_value_type;
+    using vertex_user_value_allocator_type =
+        typename std::allocator_traits<Allocator>::template rebind_alloc<vertex_user_value_type>;
+    using vertex_user_value_set =
+        detail::graph_container<vertex_user_value_type, vertex_user_value_allocator_type>;
+
+    // edge types
```
Comment (andreynv) [inline cpp/oneapi/dal/data/undirected_adjacency_array_graph.hpp:61]: minor: what should be the style of the comments?

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.hpp [hunk 2]
```diff
@@ -103,36 +113,29 @@ class ONEAPI_DAL_EXPORT undirected_adjacency_array_graph {
...
-ONEAPI_DAL_EXPORT auto get_vertex_neighbors_impl(const G &g, const vertex_type<G> &vertex) noexcept
-    -> const_vertex_edge_range_type<G>;
+template <typename Graph>
+ONEAPI_DAL_EXPORT auto get_vertex_count_impl(const Graph& graph) noexcept
+    -> vertex_size_type<Graph>;
 
-template <typename G>
-ONEAPI_DAL_EXPORT void convert_to_csr_impl(const edge_list<vertex_type<G>> &edges, G &g);
+template <typename Graph>
+ONEAPI_DAL_EXPORT auto get_edge_count_impl(const Graph& graph) noexcept -> edge_size_type<Graph>;
```
Comment (andreynv) [inline cpp/oneapi/dal/data/undirected_adjacency_array_graph.hpp:126]: why is this still here?

File: cpp/oneapi/dal/util/load_graph.hpp [hunk 3]
```diff
@@ -33,5 +33,6 @@ template <typename Descriptor>
 using output_type = typename Descriptor::output_type;
 
 template <typename Descriptor = load_graph::descriptor<>, typename DataSource = csv_data_source>
-output_type<Descriptor> load(const Descriptor &desc, const DataSource &data_source);
+ONEAPI_DAL_EXPORT output_type<Descriptor> load(const Descriptor &desc,
```
Comment (andreynv) [inline cpp/oneapi/dal/util/load_graph.hpp:36]: why is it still in utils?
Why the previous declaration via function template was removed?

File: cpp/oneapi/dal/util/csv_data_source.hpp [hunk 4]
```diff
@@ -20,7 +20,7 @@
 
 namespace oneapi::dal::preview {
 
-class csv_data_source {
+class ONEAPI_DAL_EXPORT csv_data_source {
```
Comment (andreynv) [inline cpp/oneapi/dal/util/csv_data_source.hpp:23]: why is this file in utils?

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp [hunk 5]
```diff
@@ -35,62 +35,65 @@ template class ONEAPI_DAL_EXPORT undirected_adjacency_array_graph<empty_value,
...
                                                                   std::int32_t,
-                                                                  std::allocator<empty_value>>;
+                                                                  std::allocator<char>>;
 
-using graph32 = undirected_adjacency_array_graph<empty_value,
-                                                 empty_value,
-                                                 empty_value,
-                                                 std::int32_t,
-                                                 std::allocator<empty_value>>;
+using graph_default = undirected_adjacency_array_graph<empty_value,
```
Comment (andreynv) [inline cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp:40]: what will be the name for 64 bit version of the graph?

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp [hunk 6]
```diff
@@ -35,62 +35,65 @@ template class ONEAPI_DAL_EXPORT undirected_adjacency_array_graph<empty_value,
...
     const auto &layout = detail::get_impl(g);
     return layout->_vertexes[vertex + 1] - layout->_vertexes[vertex];
 }
 
-template ONEAPI_DAL_EXPORT vertex_edge_size_type<graph32> get_vertex_degree_impl(
-    const graph32 &g,
-    const vertex_type<graph32> &vertex);
+template ONEAPI_DAL_EXPORT vertex_edge_size_type<graph_default> get_vertex_degree_impl(
+    const graph_default &g,
+    const vertex_type<graph_default> &vertex);
```
Comment (andreynv) [inline cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp:74]: can you do one style for & and ' ' across graph files?

File: cpp/oneapi/dal/data/graph_service_functions.hpp [hunk 7]
```diff
@@ -15,12 +15,6 @@
  * limitations under the License.
  *******************************************************************************/
 
-/*
-//++
-//  Graph types and service functionality
-//--
-*/
```
Comment (andreynv) [inline cpp/oneapi/dal/data/graph_service_functions.hpp:22]: why was it removed? what is the style ?

File: cpp/oneapi/dal/data/graph_service_functions.hpp [hunk 8]
```diff
@@ -36,46 +30,46 @@ namespace oneapi::dal::preview {
...
+/// @tparam Graph  Type of graph
 ///
-/// @param [in]   g  graph
+/// @param [in]   graph  Input graph
 ///
-/// @return Result is vertex_size_type of G.
-template <typename G>
-constexpr auto get_vertex_count(const G &g) noexcept -> vertex_size_type<G> {
-    return get_vertex_count_impl(g);
+/// @return Result is vertex_size_type of Graph.
```
Comment (andreynv) [inline cpp/oneapi/dal/data/graph_service_functions.hpp:37]: sounds a bit strange - return type of Graph
a routine returns  'a value' of some type, this function returns vertex count for the graph

File: cpp/oneapi/dal/data/graph_service_functions.hpp [hunk 9]
```diff
@@ -36,46 +30,46 @@ namespace oneapi::dal::preview {
...
-    return get_edge_count_impl(g);
+template <typename Graph>
+constexpr auto get_edge_count(const Graph& graph) noexcept -> edge_size_type<Graph> {
+    return get_edge_count_impl(graph);
 }
 
 /// Get degree of vertex in a graph.
 ///
-/// @tparam G  Type of graph
+/// @tparam Graph  Type of graph
```
Comment (andreynv) [inline cpp/oneapi/dal/data/graph_service_functions.hpp:50]: 'term' type may not be clean and mixed with type in programming sense
What term do we use to describe directed, undirected, represented with adjacency list etc graphs?

File: cpp/oneapi/dal/data/detail/undirected_adjacency_array_graph_impl.hpp [hunk 10]
```diff
@@ -23,11 +23,21 @@
 
 namespace oneapi::dal::preview::detail {
```
Comment (andreynv) [inline cpp/oneapi/dal/data/detail/undirected_adjacency_array_graph_impl.hpp:24]: what is the style?
namespace oneapi:...::detail or
namespace oneapi {
namespace dal {
}
}

File: cpp/oneapi/dal/algo/jaccard/backend/cpu/select_kernel.cpp [hunk 11]
```diff
@@ -32,12 +32,12 @@ similarity_result backend_default<Float, Method, Graph>::operator()(
 }
 
 template <>
-dal::detail::pimpl<backend_base<undirected_adjacency_array<> &>>
-get_backend<float, method::by_default, undirected_adjacency_array<> &>(
+dal::detail::pimpl<backend_base<undirected_adjacency_array_graph<> &>>
```
Comment (andreynv) [inline cpp/oneapi/dal/algo/jaccard/backend/cpu/select_kernel.cpp:35]: minor: would it make sense to use shorter alias for this long name in the implementation?

File: cpp/oneapi/dal/data/graph_service_functions.hpp (hunk 7 above)
Comment (bysheva) [reply cpp/oneapi/dal/data/graph_service_functions.hpp:22]: Documentation will be updated in separate pull request. The style removed here was incorrect

File: cpp/oneapi/dal/data/graph_service_functions.hpp (hunk 8 above)
Comment (bysheva) [reply cpp/oneapi/dal/data/graph_service_functions.hpp:37]: Documentation will be updated in separate pull request.

File: cpp/oneapi/dal/data/graph_service_functions.hpp (hunk 9 above)
Comment (bysheva) [reply cpp/oneapi/dal/data/graph_service_functions.hpp:50]: Documentation will be updated in separate pull request.

File: cpp/oneapi/dal/util/load_graph.hpp (hunk 3 above)
Comment (bysheva) [reply cpp/oneapi/dal/util/load_graph.hpp:36]: ONEAPI_DAL_EXPORT was added. 
The file will be moved from utils

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.hpp (hunk 2 above)
Comment (orrrrtem) [reply cpp/oneapi/dal/data/undirected_adjacency_array_graph.hpp:126]: waiting answer from dal team.

File: cpp/oneapi/dal/util/load_graph.hpp (hunk 3 above)
Comment (orrrrtem) [reply cpp/oneapi/dal/util/load_graph.hpp:36]: waiting answer from dal team.

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp (hunk 5 above)
Comment (orrrrtem) [reply cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp:40]: no current solution for it. graph64 or graph_with_int64_t_index_type??? Note: these usings are only for compact instances. they are not exposed to public.

File: cpp/oneapi/dal/data/detail/undirected_adjacency_array_graph_impl.hpp (hunk 10 above)
Comment (orrrrtem) [reply cpp/oneapi/dal/data/detail/undirected_adjacency_array_graph_impl.hpp:24]: any

File: cpp/oneapi/dal/util/load_graph.hpp (hunk 3 above)
Comment (orrrrtem) [reply cpp/oneapi/dal/util/load_graph.hpp:27]: changed for header based impl.

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp (hunk 5 above)
Comment (andreynv) [reply cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp:40]: can we discuss and agree offline on the name

File: cpp/oneapi/dal/data/detail/undirected_adjacency_array_graph_impl.hpp (hunk 10 above)
Comment (andreynv) [reply cpp/oneapi/dal/data/detail/undirected_adjacency_array_graph_impl.hpp:24]: how is the rule formulated?

**Pull request merged**